src/vehicle.cpp

```diff
@@ -1316,41 +1310,108 @@ void vehicle::use_controls()
+            }
+        }
+
+        if( einfo.fuel_type == fuel_type_gasoline && dmg > 0.75 && one_in( 20 ) ) {
+            backfire( e );
+        }
+
+        // Damaged engines have a chance of failing to start
+        if( x_in_y( dmg * 100, 120 - (20 * cold_factor) ) ) {
+            if( one_in( 2 ) ) {
+                add_msg( _("The %s makes a deep clunking sound."), einfo.name.c_str() );
+            } else {
+                add_msg( _("The %s makes a terrible clanking sound."), einfo.name.c_str() );
+            }
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool vehicle::start_engines()
 {
-    bool failed_start = false;
-    bool muscle_powered = false;
-    // TODO: Make chance of success based on engine condition.
-    // electric and plasma engines don't require anything special
+    int attempted = 0;
+    int started = 0;
+    int not_muscle = 0;
+
     for( size_t e = 0; e < engines.size(); ++e ) {
         if( is_engine_on( e ) ) {
-            if( is_engine_type_on(e, fuel_type_gasoline)  ||
-                is_engine_type_on(e, fuel_type_diesel) ) {
-                // Big engines can't be pull-started
-                int engine_power = part_power(engines[e]);
-                if(engine_power >= 50) {
-                    // Starter motor battery draw proportional to engine power
-                    if(discharge_battery(engine_power / 10)) {
-                        failed_start = true;
-                    }
-                }
-            } else if (is_engine_type_on(e, fuel_type_muscle)) {
-                muscle_powered = true;
-            }
-        } else if( is_part_on( engines[e] ) ) {
-            failed_start = true;
+            attempted++;
+            if( start_engine( e ) ) { started++; }
+            if( !is_engine_type( e, fuel_type_muscle ) ) { not_muscle++; }
         }
     }
 
-    if (failed_start) {
-        add_msg (_("The %s's engine fails to start."), name.c_str());
-    } else {
-        if (!muscle_powered) {
-            add_msg(_("The %s's engine starts up."), name.c_str());
+    if( attempted == 0 ) {
+        add_msg( _("The %s has no engine."), name.c_str() );
+    } else if( not_muscle > 0 ) {
+        if( started == attempted ) {
+            add_msg( ngettext("The %s's engine starts up.", "The %s's engines start up.", not_muscle), name.c_str() );
+        } else {
+            add_msg( ngettext("The %s's engine fails to start.", "The %s's engines fail to start.", not_muscle), name.c_str() );
         }
-        //turn on engine since nothing bad happened
-        engine_on = true;
     }
+
+    engine_on = started == attempted;
+    return engine_on;
+}
+
+void vehicle::backfire( const int e )
+{
+    const int power = part_power( engines[e], true );
+    const point p = parts[engines[e]].mount;
+    int gx, gy;
+    coord_translate( p.x, p.y, gx, gy );
+
+    sounds::ambient_sound( global_x() + gx, global_y() + gy, 40 + (power / 30), "BANG!" );
```

Comment:
You can take the coordinates from the vehicle part directly without having them translated (`precalc[0]` is the location of the part in map coordinates, still relative to the vehicles position, but they already include the vehicle rotation):
```C++
point const pos = global_pos() + parts[p].precalc[0];
```

It's done similar (but with separate x,y values) in `fire_turret_internal`:
```C++
bool vehicle::fire_turret_internal (int p, const itype &gun, const itype &ammo, long &charges, const std::string &extra_sound)
{
    int x = global_x() + parts[p].precalc[0].x;
    int y = global_y() + parts[p].precalc[0].y;
    ...
    sounds::sound(x, y, 20, extra_sound);
```